NPE in user task import

### Describe the bug

<!-- A clear and concise description of what the problem/bug is about and what is the current behavior. -->
Such exception was discovered in code:
```
io.camunda.operate.exceptions.PersistenceException: java.lang.NullPointerException: Cannot invoke "io.camunda.operate.util.Tuple.getLeft()" because "fieldAndValueSupplier" is null
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.performImport(AbstractImportBatchProcessor.java:54) ~[operate-importer-common-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.ImportJob.processOneIndexBatch(ImportJob.java:161) ~[operate-importer-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.ImportJob.call(ImportJob.java:80) ~[operate-importer-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.elasticsearch.ElasticsearchRecordsReader.lambda$executeJob$5(ElasticsearchRecordsReader.java:561) ~[operate-importer-8.5.0.jar!/:8.5.0]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "io.camunda.operate.util.Tuple.getLeft()" because "fieldAndValueSupplier" is null
	at io.camunda.operate.zeebeimport.v8_5.processors.UserTaskZeebeRecordProcessor.getUpdateFields(UserTaskZeebeRecordProcessor.java:111) ~[operate-importer-8_5-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.v8_5.processors.UserTaskZeebeRecordProcessor.processUserTaskRecord(UserTaskZeebeRecordProcessor.java:145) ~[operate-importer-8_5-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.v8_5.processors.ImportBulkProcessor.processUserTask(ImportBulkProcessor.java:315) ~[operate-importer-8_5-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.v8_5.processors.ImportBulkProcessor.processZeebeRecords(ImportBulkProcessor.java:153) ~[operate-importer-8_5-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.lambda$performImport$0(AbstractImportBatchProcessor.java:41) ~[operate-importer-common-8.5.0.jar!/:8.5.0]
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175) ~[micrometer-core-1.12.4.jar!/:1.12.4]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.withTimer(AbstractImportBatchProcessor.java:78) ~[operate-importer-common-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.withProcessingTimer(AbstractImportBatchProcessor.java:60) ~[operate-importer-common-8.5.0.jar!/:8.5.0]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.performImport(AbstractImportBatchProcessor.java:39) ~[operate-importer-common-8.5.0.jar!/:8.5.0]
	... 7 more
```

### To Reproduce

Steps to reproduce the behavior:
TBD

### Current behavior

<!-- Please attach screenshots, a screen recording, or a file (e.g. the BPMN/DMN/Form file) that has the problem you are describing to help us better debug the respective issue. -->
Import is blocked with NPE

### Expected behavior

<!-- A clear and concise description of what you expected to happen. -->
No NPE, import progresses

### Environment

<!-- Please provide details about the environment you were in when the problem occurred. -->
Operate Version: 8.5.0

### Additional context

<!-- Please add any other context about the problem. Here you can also provide us some data that you used while the bug happen like **json** file or specific **BPMN**. -->
* related incident: https://camunda.slack.com/archives/C06UH8S9JQP
* processing of the following Zeebe record caused the exception:
```
{
                "_index": "zeebe-record_user-task_8.5.0_2024-04-18",
                "_id": "1-275287",
                "_score": null,
                "_routing": "1",
                "_source": {
                    "partitionId": 1,
                    "value": {
                        "processInstanceKey": 2251799813822626,
                        "elementInstanceKey": 2251799813822630,
                        "tenantId": "<default>",
                        "bpmnProcessId": "Process_0a32bx6",
                        "processDefinitionKey": 2251799813822624,
                        "processDefinitionVersion": 1,
                        "elementId": "Activity_0fluidy",
                        "customHeaders": {},
                        "candidateGroupsList": [
                            "string"
                        ],
                        "candidateUsersList": [
                            "string"
                        ],
                        "creationTimestamp": 1713425582479,
                        "changedAttributes": [
                            "candidateGroupsList",
                            "candidateUsersList",
                            "dueDate",
                            "followUpDate"
                        ],
                        "externalFormReference": "",
                        "dueDate": "2024-04-03T20:22:18.305Z",
                        "userTaskKey": 2251799813822631,
                        "formKey": 2251799813822623,
                        "followUpDate": "2024-04-03T20:22:18.305Z",
                        "variables": {},
                        "action": "string",
                        "assignee": ""
                    },
                    "key": 2251799813822631,
                    "timestamp": 1713425730856,
                    "valueType": "USER_TASK",
                    "brokerVersion": "8.5.0",
                    "recordType": "EVENT",
                    "intent": "UPDATED",
                    "authorizations": {
                        "authorized_tenants": [
                            "<default>"
                        ]
                    },
                    "recordVersion": 1,
                    "sourceRecordPosition": 275285,
                    "rejectionType": "NULL_VAL",
                    "rejectionReason": "",
                    "position": 275287,
                    "sequence": 2251799813685252
                },
                "sort": [
                    2251799813685252
                ]
            }
```
* IMO the reason is in how `candidateGroupsList` and `candidateUsersList` fields are named, [in Operate code](https://github.com/camunda/zeebe/blob/8bd0c74ff7b98c192721cbcbfb9b290e1d1c43b6/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/v8_5/processors/UserTaskZeebeRecordProcessor.java#L96-L98) we use singular form
---------------------------------------------------------------------------------------------

<!-- As the creator of the issue, you don't have to fill anything below this line, but the assignee will take care of this as part. -->

### Acceptance Criteria

<!-- the assignee will fill the Acceptance Criteria. -->

### Definition of Ready - Checklist

<!-- the assignee will check the DOR. -->
- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### Implementation

#### :mag: Root Cause Analysis

<!-- Explain the underlying cause for the issue. -->

#### :thought_balloon: Proposed Solution

<!-- Provide a high level overview of the proposed solution. (Technical details will be available in the associated PR) -->

### :point_right: Handover Dev to QA

<!--As a team, we have settled on a checklist to remind the DRI what information to provide to help the QA Engineer perform a frictionless and targeted QA test. The information requested by the checklist can be added before review/moving the ticket to the QA test column as a comment on the ticket.-->
- Changed components:
- Side effects on other components:
- Handy resources:
  BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc :
  <!-- Add here -->
- Example projects:

<!-- Add here -->
- Commands/Steps needed to test; Versions to validate:

<!-- Add here -->
- Docker file / HELM chart : in case that it needed to be tested via docker share the version contain the fixed along with version of other services .

<!--elasticsearch: 16.2.2
identitiy:alpha3
zeebe:alpha3
Operate:alpha3
tasklist:alpha3-->
- Release version ( in which version this fixed/feature will be released):

<!-- Add here -->

### :green_book: Link to the test case

<!-- please add test case link for this bug if there is any if not after testing QA will  create a test case for it and add it here. -->

--- a/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/v8_5/processors/UserTaskZeebeRecordProcessor.java
+++ b/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/v8_5/processors/UserTaskZeebeRecordProcessor.java
@@ -95,7 +95,11 @@ public class UserTaskZeebeRecordProcessor {
         Map.of(
             "candidateUserList",
             new Tuple<>(UserTaskTemplate.CANDIDATE_USERS, userTaskEntity::getCandidateUsers),
+            "candidateUsersList",
+            new Tuple<>(UserTaskTemplate.CANDIDATE_USERS, userTaskEntity::getCandidateUsers),
             "candidateGroupList",
+            new Tuple<>(UserTaskTemplate.CANDIDATE_GROUPS, userTaskEntity::getCandidateGroups),
+            "candidateGroupsList",
             new Tuple<>(UserTaskTemplate.CANDIDATE_GROUPS, userTaskEntity::getCandidateGroups),
             "dueDate",
             new Tuple<>(UserTaskTemplate.DUE_DATE, userTaskEntity::getDueDate),
@@ -108,7 +112,14 @@ public class UserTaskZeebeRecordProcessor {
     map.put(UserTaskTemplate.ACTION, userTaskEntity.getAction());
     for (final var changedAttribute : changedAttributes) {
       final var fieldAndValueSupplier = changedAttributesToUserEntitySupplier.get(changedAttribute);
-      map.put(fieldAndValueSupplier.getLeft(), fieldAndValueSupplier.getRight().get());
+      if (fieldAndValueSupplier != null) {
+        map.put(fieldAndValueSupplier.getLeft(), fieldAndValueSupplier.getRight().get());
+      } else {
+        LOGGER.warn(
+            "Could not find attribute {} from changed attributes {}. This will be ignored.",
+            changedAttribute,
+            changedAttributes);
+      }
     }
     return map;
   }

--- a/operate/importer-8_5/src/test/java/io/camunda/operate/zeebeimport/v8_5/processors/UserTaskZeebeRecordProcessorTest.java
+++ b/operate/importer-8_5/src/test/java/io/camunda/operate/zeebeimport/v8_5/processors/UserTaskZeebeRecordProcessorTest.java
@@ -202,6 +202,56 @@ class UserTaskZeebeRecordProcessorTest {
     verify(batchRequest).update("user-task-index", "1", expectedUpdateFields);
   }
 
+  @Test // https://github.com/camunda/zeebe/issues/17611
+  void updateShouldIgnoreMissingAttributes() throws PersistenceException {
+    /* given */
+    final var batchRequest = mock(BatchRequest.class);
+    final var userTaskRecord = (Record<UserTaskRecordValue>) mock(Record.class);
+    final var userTaskRecordValue =
+        ImmutableUserTaskRecordValue.builder()
+            .withUserTaskKey(1L)
+            .withTenantId(DEFAULT_TENANT_ID)
+            .withCandidateUsersList(List.of("string"))
+            .withCandidateGroupsList(List.of("string"))
+            .withDueDate("2024-04-03T20:22:18.305Z")
+            .withFollowUpDate("2024-04-03T20:22:18.305Z")
+            .withAction("update")
+            .withChangedAttributes(
+                List.of(
+                    "candidateUsersList",
+                    "candidateGroupsList",
+                    "dueDate",
+                    "followUpDate",
+                    "attribute-not-exist"))
+            .build();
+    when(userTaskTemplate.getFullQualifiedName()).thenReturn("user-task-index");
+    when(userTaskRecord.getIntent()).thenReturn(UserTaskIntent.UPDATED);
+    when(userTaskRecord.getValue()).thenReturn(userTaskRecordValue);
+    /* when */
+    userTaskZeebeRecordProcessor.processUserTaskRecord(batchRequest, userTaskRecord);
+    /* then */
+    final Map<String, Object> expectedUpdateFields =
+        Map.of(
+            "candidateUsers",
+            List.of("string"),
+            "candidateGroups",
+            List.of("string"),
+            "dueDate",
+            OffsetDateTime.parse("2024-04-03T20:22:18.305Z"),
+            "followUpDate",
+            OffsetDateTime.parse("2024-04-03T20:22:18.305Z"),
+            "action",
+            "update",
+            "changedAttributes",
+            List.of(
+                "candidateUsersList",
+                "candidateGroupsList",
+                "dueDate",
+                "followUpDate",
+                "attribute-not-exist"));
+    verify(batchRequest).update("user-task-index", "1", expectedUpdateFields);
+  }
+
   @Test
   void completedEvent() throws PersistenceException, JsonProcessingException {
     /* given */

--- a/operate/importer-8_6/src/main/java/io/camunda/operate/zeebeimport/v8_6/processors/UserTaskZeebeRecordProcessor.java
+++ b/operate/importer-8_6/src/main/java/io/camunda/operate/zeebeimport/v8_6/processors/UserTaskZeebeRecordProcessor.java
@@ -95,7 +95,11 @@ public class UserTaskZeebeRecordProcessor {
         Map.of(
             "candidateUserList",
             new Tuple<>(UserTaskTemplate.CANDIDATE_USERS, userTaskEntity::getCandidateUsers),
+            "candidateUsersList",
+            new Tuple<>(UserTaskTemplate.CANDIDATE_USERS, userTaskEntity::getCandidateUsers),
             "candidateGroupList",
+            new Tuple<>(UserTaskTemplate.CANDIDATE_GROUPS, userTaskEntity::getCandidateGroups),
+            "candidateGroupsList",
             new Tuple<>(UserTaskTemplate.CANDIDATE_GROUPS, userTaskEntity::getCandidateGroups),
             "dueDate",
             new Tuple<>(UserTaskTemplate.DUE_DATE, userTaskEntity::getDueDate),
@@ -108,7 +112,14 @@ public class UserTaskZeebeRecordProcessor {
     map.put(UserTaskTemplate.ACTION, userTaskEntity.getAction());
     for (final var changedAttribute : changedAttributes) {
       final var fieldAndValueSupplier = changedAttributesToUserEntitySupplier.get(changedAttribute);
-      map.put(fieldAndValueSupplier.getLeft(), fieldAndValueSupplier.getRight().get());
+      if (fieldAndValueSupplier != null) {
+        map.put(fieldAndValueSupplier.getLeft(), fieldAndValueSupplier.getRight().get());
+      } else {
+        LOGGER.warn(
+            "Could not find attribute {} from changed attributes {}. This will be ignored.",
+            changedAttribute,
+            changedAttributes);
+      }
     }
     return map;
   }

--- a/operate/importer-8_6/src/test/java/io/camunda/operate/zeebeimport/v8_6/processors/UserTaskZeebeRecordProcessorTest.java
+++ b/operate/importer-8_6/src/test/java/io/camunda/operate/zeebeimport/v8_6/processors/UserTaskZeebeRecordProcessorTest.java
@@ -179,7 +179,7 @@ class UserTaskZeebeRecordProcessorTest {
             .withCandidateGroupsList(List.of("Simpsons", "Flanders"))
             .withDueDate("2023-05-23T01:02:03+01:00")
             .withAction("update")
-            .withChangedAttributes(List.of("candidateUserList", "candidateGroupList", "dueDate"))
+            .withChangedAttributes(List.of("candidateUsersList", "candidateGroupsList", "dueDate"))
             .build();
     when(userTaskTemplate.getFullQualifiedName()).thenReturn("user-task-index");
     when(userTaskRecord.getIntent()).thenReturn(UserTaskIntent.UPDATED);
@@ -198,7 +198,57 @@ class UserTaskZeebeRecordProcessorTest {
             "action",
             "update",
             "changedAttributes",
-            List.of("candidateUserList", "candidateGroupList", "dueDate"));
+            List.of("candidateUsersList", "candidateGroupsList", "dueDate"));
+    verify(batchRequest).update("user-task-index", "1", expectedUpdateFields);
+  }
+
+  @Test // https://github.com/camunda/zeebe/issues/17611
+  void updateShouldIgnoreMissingAttributes() throws PersistenceException {
+    /* given */
+    final var batchRequest = mock(BatchRequest.class);
+    final var userTaskRecord = (Record<UserTaskRecordValue>) mock(Record.class);
+    final var userTaskRecordValue =
+        ImmutableUserTaskRecordValue.builder()
+            .withUserTaskKey(1L)
+            .withTenantId(DEFAULT_TENANT_ID)
+            .withCandidateUsersList(List.of("string"))
+            .withCandidateGroupsList(List.of("string"))
+            .withDueDate("2024-04-03T20:22:18.305Z")
+            .withFollowUpDate("2024-04-03T20:22:18.305Z")
+            .withAction("update")
+            .withChangedAttributes(
+                List.of(
+                    "candidateUsersList",
+                    "candidateGroupsList",
+                    "dueDate",
+                    "followUpDate",
+                    "attribute-not-exist"))
+            .build();
+    when(userTaskTemplate.getFullQualifiedName()).thenReturn("user-task-index");
+    when(userTaskRecord.getIntent()).thenReturn(UserTaskIntent.UPDATED);
+    when(userTaskRecord.getValue()).thenReturn(userTaskRecordValue);
+    /* when */
+    userTaskZeebeRecordProcessor.processUserTaskRecord(batchRequest, userTaskRecord);
+    /* then */
+    final Map<String, Object> expectedUpdateFields =
+        Map.of(
+            "candidateUsers",
+            List.of("string"),
+            "candidateGroups",
+            List.of("string"),
+            "dueDate",
+            OffsetDateTime.parse("2024-04-03T20:22:18.305Z"),
+            "followUpDate",
+            OffsetDateTime.parse("2024-04-03T20:22:18.305Z"),
+            "action",
+            "update",
+            "changedAttributes",
+            List.of(
+                "candidateUsersList",
+                "candidateGroupsList",
+                "dueDate",
+                "followUpDate",
+                "attribute-not-exist"));
     verify(batchRequest).update("user-task-index", "1", expectedUpdateFields);
   }
 

